src/build.jl

```diff
@@ -103,3 +120,73 @@ function _default_preamble()
     end
     return preamble
 end
+
+#######
+# PNG # 
+#######
+@enum(PNGEngine, NO_PNG_ENGINE, PNG_PDF_TO_CAIRO, PDF_TO_PPM)
+const ACTIVE_PNG_ENGINE = Ref{Union{Nothing, PNGEngine}}(nothing)
+function png_engine()
```

Comment:
`latex_engine`, `svg_engine` and `png_engine` are basically the same code. Could be factored out into something like
```julia
function find_engine!(kind, engine, executable_enum_pairs, fallback_enum; error = nothing)
    if engine[] ≢ nothing
        for (executable, enum) in engine_enum_pairs
            @debug "$(kind)_engine: looking for $(kind) engine $(executable)"
            if Sys.which(executable) ≢ nothing
                @debug "$(kind)_engine: found $(kind) engine $(executable), using it"
                return engine[] = enum
            end
        end
        if error ≢ nothing
            throw(error)
        else
            return engine[] = fallback_enum
        end
    end
    engine[]
end
```
and called by all 3 functions.

Reply:
I thought about that but I felt that it wasn't really at the level where factoring it out would be worth it compared to the overhead of the extra "abstractness".